states/battle.py

```diff
@@ -28,30 +30,126 @@ class Battle(State):
     def __init__(self):
         super().__init__()
         self.enemy = asyncio.run(pokeapi.get_random_pokemon_from_api())
-        is_escape = False
+        end_encounter = False
+
         print(MESSAGES["BATTLE"]["ENTRY"].format(self.enemy.name), end="\n" * 2)
-        print(f"Go {PLAYER.lead_pokemon.name}! \n")
-        while is_escape is not True:
+        self.send_pokemon(PLAYER.lead_pokemon)
+        self.print_health(PLAYER.lead_pokemon)
+
+        while end_encounter is not True:
             self.option = self.check_input(BattleOptions)
             # TODO: Implement battling
             if self.option == BattleOptions.FIGHT:
                 pass
             elif self.option == BattleOptions.TEAM:
                 pass
             elif self.option == BattleOptions.CATCH:
-                pass
+                if self.calc_catch_prob() is True:
+                    self.catch_pokemon()
+                    end_encounter = True
+                else:
+                    print(MESSAGES["BATTLE"]["CATCH_FAIL"].format(self.enemy.name))
+                    self.enemy_pokemon_attack()
+                    self.health_case(
+                        self.check_health_zero(PLAYER.lead_pokemon),
+                        PLAYER.lead_pokemon,
+                    )
             elif self.option == BattleOptions.RUN:
-                is_escape = self.calc_run_prob()
+                end_encounter = self.calc_run_prob()
```

Comment:
This too

Reply:
Created `run_fail()` to handle this functionality